File: Week_03/226/invert_binary_tree2.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        queue = [root]
        while queue:
            node = queue.pop()
            if node:
                node.right, node.left = node.left, node.right
                queue.append(node.right)
                queue.append(node.left)
        return root

File: Week_03/226/invert_binary_tree2.go
package main

func invertTree2(root *TreeNode) *TreeNode {
	queue := []*TreeNode{root}
	for len(queue) > 0 {
		node := queue[0]
		queue = queue[1:]
		if node != nil {
			node.Left, node.Right = node.Right, node.Left
			queue = append(queue, node.Left, node.Right)
		}
	}
	return root
}
